qcodes/dataset/plotting.py

```diff
@@ -269,6 +288,110 @@ def plot_by_id(run_id: int,
+                                                  degrees=degrees)
+                if index < len(group) - 1:
+                    # if the above condition is met, we are dealing with
+                    # complex setpoints
+                    new_group.append(p1)
+                    new_group.append(p2)
+                else:
+                    # in this case, we are dealing with a complex top-level
+                    # parameter. Also, all the setpoints will have been handled
+                    # by now. We split the group into two groups, one for each
+                    # new (real) top-level parameter
+                    new_groups[0] = new_group.copy()
+                    new_groups[1] = new_group.copy()
+                    new_groups[0].append(p1)
+                    new_groups[1].append(p2)
+            else:
+                new_group.append(parameter)
+        if new_groups == [[], []]:
+            # if the above condition is met, the group did not contain a
+            # complex top-level parameter and has thus not been split into two
+            # new groups
+            newdata.append(new_group)
+        else:
+            newdata.append(new_groups[0])
+            newdata.append(new_groups[1])
+
+    return newdata
+
+
+def _convert_complex_to_real(
+        parameter: Dict[str, Union[str, np.ndarray]],
+        conversion: str,
+        degrees: bool
+        ) -> Tuple[Dict[str, Union[str, np.ndarray]],
+                   Dict[str, Union[str, np.ndarray]]]:
+    """
+    Do the actual conversion and turn one parameter into two.
+    Should only be called from within _complex_to_real_preparser.
+    """
+
+    phase_unit = 'deg' if degrees else 'rad'
+
+    converters = {
+        'data': {'real_and_imag': lambda x: (np.real(x), np.imag(x)),
+                 'mag_and_phase': lambda x: (np.abs(x),
+                                             np.angle(x, deg=degrees))},
+        'labels': {'real_and_imag': lambda l: (l + ' [real]', l + ' [imag]'),
+                   'mag_and_phase': lambda l: (l + ' [mag]', l + ' [phase]')},
+        'units': {'real_and_imag': lambda u: (u, u),
+                  'mag_and_phase': lambda u: (u, phase_unit)},
+        'names': {'real_and_imag': lambda n: (n + '_real', n + '_imag'),
+                  'mag_and_phase': lambda n: (n + '_mag', n + '_phase')}}
+
+    new_data = converters['data'][conversion](parameter['data'])
+    new_labels = converters['labels'][conversion](parameter['label'])
+    new_units = converters['units'][conversion](parameter['unit'])
+    new_names = converters['names'][conversion](parameter['name'])
+
+    new_parameters: List[Dict[str, Union[str, np.ndarray]]] = []
+
+    for n in range(2):
+        d: Dict[str, Union[str, np.ndarray]]
+        d = {'name': new_names[n], 'label': new_labels[n],
+             'unit': new_units[n], 'data': new_data[n]}
+        new_parameters.append(d)
+
+    return tuple(new_parameters)  # type: ignore
+
```

Comment:
may i suggest a subjectively simpler implementation of this logic?:
```
new_parameters = (
    {'name': name, 'label': label,
     'unit': unit, 'data': data}
    for name, label, unit, data in zip(
        new_names, new_labels, new_units, new_data)
)

return tuple(new_parameters)
```

Reply:
I was hoping that all the extra type annotations would somehow be helpful. I guess they're not?

Reply:
they are but the `for n in range(2):` part kind of kills it, at least for me. Hence, the suggestion.